src/log_uploader.js

```diff
@@ -59,24 +96,61 @@ function upload_ask() {
 }
 
 function upload() {
-	const github_log_repo = config.github_log_repo;
-	const shouldUploadWithGithub = github_log_repo != null && github_log_repo != '';
+	return shouldUploadWithS3 ? uploadToS3() : (shouldUploadWithGithub ? uploadToGithub() : uploadToSpringRTS());
+}
+
+function uploadToS3() {
+	const fileContent = fs.readFileSync(logPath);
+	const fileName = 'infolog_' + new Date().getTime() + '.txt' // File name to save as
+
+	const s3 = new AWS.S3({
+		accessKeyId: aws_id,
+		secretAccessKey: aws_secret
+	});
 
-	return shouldUploadWithGithub ? uploadToGithub() : uploadToSpringRTS();
+	const params = {
+		Bucket: logs_s3_bucket,
+		Key: fileName,
+		Body: fileContent,
+		ContentType: 'text/plain'
+	};
+
+	return new Promise((resolve, reject) => {
+		s3.upload(params, function(err, data) {
+			if (err) {
+				reject(err);
+			}
+			resolve({ 'url': data?.Location });
+		});
+	});
 }
 
 function uploadToGithub() {
-	const github_client = github.client(github_access_token);
-	const ghrepo = github_client.repo('Spring-SpringBoard/Logs');
-	return ghrepo.issueAsync({
-		'title': `spring-launcher log: ${config.title}`,
-		'body': '```' + fs.readFileSync(logPath).toString() + '```',
-	}).then(obj => {
-		return { 'url': obj[0].html_url };
+	const fileName = 'infolog_' + new Date().getTime() + '.txt' // File name to save as
+
+	var content = {
+		description: '',
+		files: {
+			[fileName]: {
+				'content': fs.readFileSync(logPath).toString(),
+			}
+		}
+	};
+
+	const gistClient = new GistClient()
+
+	gistClient.setToken(github_access_token)
+
+	return new Promise((resolve, reject) => {
+		gistClient.create(content).then(newGist => {
+			resolve({ 'url': newGist?.html_url });
+		}).catch(err => {
+			reject(err.toString());
+		});
 	});
```

Comment:
Why would we change from GH repo to gists?

Reply:
Issues are public, Gists are private and are only visible when the URL is known. Logs might contain users; personal information, so it's best to give them control over whom to share it with.
Also, creating an issue for every log upload is too much noise in the repository...

Reply:
Repos can be private, although that would make the posted issue invisible even to the person posting. So gists do make sense in that regard.

Regarding noise, you can just use a special repo for that. It's why I made https://github.com/beyond-all-reason/Logs/issues originally. The benefit of a custom repo is organization. Devs can easily search and categorize logs if they're posted in one repo.